clickhouse-admin/src/http_entrypoints.rs

```diff
@@ -146,28 +144,40 @@ impl ClickhouseAdminSingleApi for ClickhouseAdminSingleImpl {
     ) -> Result<HttpResponseUpdatedNoContent, HttpError> {
         let log = &rqctx.log;
         let ctx = rqctx.context();
-        let ip = ctx.clickhouse_cli().listen_address.ip();
-        let address = SocketAddrV6::new(*ip, CLICKHOUSE_TCP_PORT, 0, 0);
-        let client = OximeterClient::new(address.into(), log);
-        debug!(
-            log,
-            "initializing single-node ClickHouse \
-             at {address} to version {OXIMETER_VERSION}"
-        );
 
         // Database initialization is idempotent, but not concurrency-safe.
         // Use a mutex to serialize requests.
         let lock = ctx.initialization_lock();
         let _guard = lock.lock().await;
-        client
-            .initialize_db_with_version(false, OXIMETER_VERSION)
-            .await
-            .map_err(|e| {
-                HttpError::for_internal_error(format!(
-                    "can't initialize single-node ClickHouse \
-                     at {address} to version {OXIMETER_VERSION}: {e}",
-                ))
-            })?;
+
+        // Initialize the database only if it was not previously initialized.
+        // TODO: Migrate schema to newer version without wiping data.
+        let client = ctx.oximeter_client();
+        let version = client.read_latest_version().await.map_err(|e| {
+            HttpError::for_internal_error(format!(
+                "can't read ClickHouse version: {e}",
+            ))
+        })?;
+        if version == 0 {
```

Comment:
With apologies for this question possibly being more relevant to #6903 - how does this interact with Oximeter doing almost exactly (or maybe exactly exactly) this same check-if-0-and-initialize on startup, given the notes about initialization not being concurrency-safe?

Reply:
We'll take this as a follow-up: #7173.